cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.artifact;
+
+import java.io.IOException;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Provides access to artifacts
+ */
+public interface ArtifactManager {
+  /**
+   * Get the list of artifacts in the repository in the current and system namespace
+   *
+   * @return {@link List<ArtifactInfo>} list of artifact info
+   * @throws IOException when there is an error retrieving artifacts
+   */
+  List<ArtifactInfo> listArtifacts() throws IOException;
+
+  /**
+   * Create a class loader using the artifact represented by artifactInfo with parent as parentClassloader
+   * Call to this method might take long based on the size of artifact,
+   * if called from short transactions, its possible this call will timeout if artifact size is big
+   *
+   * @param artifactInfo artifact info
+   * @param parentClassLoader parent class loader, if null bootstrap classLoader shall be used as parent
+   * @throws IOException if there were any exception while creating the class loader
+   * @return Closeable class loader, calling close on this does the necessary cleanup.
+   */
+  CloseableClassLoader createClassLoader(ArtifactInfo artifactInfo,
```

Comment:
there should be clearly defined behavior if the specified artifact info does not exist. For example, throw an ArtifactNotFoundException